File: test_scripts/list_official_matches.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';
import 'package:flutter/material.dart';

void main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp();
  final db = FirebaseFirestore.instance;

  print('--- OFFICIAL MATCHES ---');
  final snap = await db
      .collection('official_leagues')
      .doc('mens-t20-world-cup-2026')
      .collection('matches')
      .get();

  for (var doc in snap.docs) {
    print(
      'OFFICIAL: ${doc.data()['team1Name']} vs ${doc.data()['team2Name']} | Time: ${doc.data()['scheduledTime']}',
    );
  }
}

File: list_official_matches.py
import json
import os

# Fetch official matches for T20 WC 2026
cmd = 'curl.exe -X GET "https://firestore.googleapis.com/v1/projects/winniko-real/databases/(default)/documents/official_leagues/mens-t20-world-cup-2026/matches?pageSize=100" -o official_matches_t20.json'
os.system(cmd)

with open('official_matches_t20.json', 'r', encoding='utf-8') as f:
    try:
        data = json.load(f)
        if 'documents' in data:
            print(f"--- Official Matches for T20 WC ---")
            for doc in data['documents']:
                fields = doc.get('fields', {})
                t1 = fields.get('team1Name', {}).get('stringValue', 'N/A')
                t2 = fields.get('team2Name', {}).get('stringValue', 'N/A')
                match_id = doc['name'].split('/')[-1]
                print(f"ID: {match_id} | {t1} vs {t2}")
        else:
            print("No matches found in official_leagues/mens-t20-world-cup-2026/matches")
            print(data)
    except Exception as e:
        print(f"Error parsing JSON: {e}")
